lib/flutter_redux.dart

```diff
@@ -107,13 +107,16 @@ typedef OnDisposeCallback<S> = void Function(
 typedef IgnoreChangeTest<S> = bool Function(S state);
 
 /// A function that will be run on State change, before the build method.
-///
+/// The same as [OnWillChangeCallback] except it provides the previous 
+/// `ViewModel`.
+/// 
 /// This function is passed the `ViewModel`, and if `distinct` is `true`,
 /// it will only be called if the `ViewModel` changes.
-///
-/// This can be useful for imperative calls to things like Navigator,
-/// TabController, etc
-typedef OnWillChangeCallback<ViewModel> = void Function(ViewModel viewModel);
+/// 
+/// This can be useful for triggering actions based on the previous 
```

Comment:
Can we combine the changed line and the new one? I don't quite want to lose that other info.

```
/// This is useful for making calls to other classes, such as a 
/// `Navigator` or `TabController`, in response to state changes. 
/// It can also be used to trigger an action based on the previous 
/// state.
```